File: readability.py
from cs50 import get_string

# Getting the text to analyze
text = get_string("Insert text here: ")

# Declaring the variables I will be keeping track of
letters = 0
spaces = 0
sentences = 0

# Using a list to look for sentences
punctuation = ['!', '?', '.']

# For loop using the len function to iterate through the text
for i in range (len(text)):
    # Looking for lower case letters
    if text[i] >= 'a' and text[i] <= 'z':
        letters += 1

    # Looking for upper case letters
    if text[i] >= 'A' and text[i] <= 'Z':
        letters += 1

    # Looking for spaces
    if text[i] == ' ':
        spaces += 1

    # Looking for punctuation
    if text[i] in punctuation:
        sentences += 1

    # Words are simply the number of spaces + 1
    words = spaces + 1

# Calculating the variables of the equation and index itself
L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Spitting out the results based on the index calculation
if index >= 16:
    print("Grade 16+")

elif index < 1:
    print("Before Grade 1")

else:
    print(f"Grade {index}")

File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>



int main(void)
{
    string text = get_string("Text: \n");
    
    // defining varriables I need to count for the index calculation
    int letters = 0;
    int spaces = 0;
    int sentences = 0;
    
    char punctuation [] = {'!', '.', '?'};
   
    // using a for loop to count letters, spaces and punctuation
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        
        // counting letters
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            letters++;
        }
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            letters++;
        }
      
        // counting spaces
        if (text[i] == ' ')
        {
            spaces++;
        }
        
        // counting sentences
        if (text[i] == punctuation[0] || text[i] == punctuation[1] || text[i] == punctuation[2])
        {
            sentences++;
        }
    }
    // words are simply the sum of spaces plus one 
    int words = spaces + 1;
   
    // using floats as the average can present decimals we don't want to overlook 
    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
   
    // double or float here is the same 
    double cl_index = round(0.0588 * L - 0.296 * S - 15.8);
   
    // using if and else if to takcle three possible outcomes 
    if (cl_index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (cl_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) cl_index);
    }
}
